easybuild/tools/modules.py

```diff
@@ -1603,12 +1619,18 @@ def get_software_version(name):
     return version
 
 
-def curr_module_paths():
+def curr_module_paths(clean=True):
```

Comment:
I wouldn't make cleaning optional at all, I think weird things may happen when empty values are included in `$MODULEPATH` and EasyBuild starts using those. There's a reason we're filtering those here...

Reply:
This is required to match Lmod behavior: Lmod uses any path as specified, so non-existing, empty or "unusual" ones are just used.
So the "fast" module use/unuse must use the non-cleaned path or behavior will differ

> There's a reason we're filtering those here...

Remember which? Also the default is still to clean, so all of EB except for the new functions is unaffected

Reply:
From Slack:
> we do need a `clean_existing_paths` for `curr_module_paths`, but it should still filter out empty paths

Why? I don't even think we should clean out non-existing paths by default and filtering out empty paths would be different from what Lmod does, although I'm not sure about the semantic of that. And if we do filter those, then the fast and regular `module.use` would be different.